modcc/symdiff.cpp

```diff
@@ -666,17 +676,20 @@ ARB_LIBMODCC_API linear_test_result linear_test(Expression* e, const std::vector
     result.constant = e->clone();
     for (const auto& id: vars) {
         auto coef = symbolic_pdiff(e, id);
-        if (!coef) {
-            return linear_test_result{};
+        if (coef->has_error()) {
+            auto res = linear_test_result{};
+            res.error({coef->error_message(), loc});
+            return res;
         }
+        if (!coef) return linear_test_result{};
         if (!is_zero(coef)) result.coef[id] = std::move(coef);
-
         result.constant = substitute(result.constant, id, zero());
     }
 
     ConstantSimplifyVisitor csimp_visitor;
     result.constant->accept(&csimp_visitor);
     result.constant = csimp_visitor.result();
+    if (result.constant.get() == nullptr) throw compiler_exception{"Simplify: linear has no constant term.", loc};
```

Comment:
why is it a problem if there is no constant term?

Reply:
Oh, that's not really the core issue. The `NULL` value is used to signal an error by the visitor. 
While the visitor has an internal error state that's pretty elaborate, it does not bubble that up 
to the user. 
 
The message could be better, though.

Reply:
gotcha, yes I think the error message should reflect that, and not state that there is no constant term